cmd/kubeadm/app/util/config/initconfiguration.go

```diff
@@ -117,10 +118,8 @@ func SetNodeRegistrationDynamicDefaults(cfg *kubeadmapi.NodeRegistrationOptions,
 		klog.V(1).Infof("detected and using CRI socket: %s", cfg.CRISocket)
 	} else {
 		if !strings.HasPrefix(cfg.CRISocket, kubeadmapiv1.DefaultContainerRuntimeURLScheme) {
-			klog.Warningf("Usage of CRI endpoints without URL scheme is deprecated and can cause kubelet errors "+
-				"in the future. Automatically prepending scheme %q to the \"criSocket\" with value %q. "+
-				"Please update your configuration!", kubeadmapiv1.DefaultContainerRuntimeURLScheme, cfg.CRISocket)
-			cfg.CRISocket = kubeadmapiv1.DefaultContainerRuntimeURLScheme + "://" + cfg.CRISocket
+			return fmt.Errorf("the CRI endpoints %s has no prefix of URL scheme %q, and this is deprecated and can cause kubelet errors in the future",
```

Comment:
I don't think "this is deprecated" is the right words still as it's a error and is prohibited now.

Reply:
https://github.com/kubernetes/kubernetes/blob/37937bb2279092ebff22eee80c64423f1f0923b2/pkg/kubelet/util/util_unix.go#L103

We mentioned it as deprecated in kubelet error message above.

Reply:
seems like `kubelet` doesn't ban the use of endpoint, but why `kubeadm` move ahead of the `kubelet`? Let's open this to other reviewers.

Reply:
That’s a good point. At the latest, we will change it from a warning to an error until kubelet does not support it.

But from the kubeadm aspect, this can be done in v1.27, as we already add the prefix `unix://` or `npipe://` in the 1.24 upgrade process. So if a user is using kubeadm and installing or upgrading to v1.27, they should use the full-url-formated annotation and endpoint. 
- This is why we can change it to error now.

Reply:
I am okay with v1.27+, and after the `kubelet`  changing it to an error as well.

Reply:
if we are making this an error it should be `pkg/errors.Errorf(....`

@chendave is making a good point, perhaps we should track what kubelet release turns these into errors and only then sync kubeadm to also throw an error.

kubelet code tends to change very slow, so unless someone starts owning the change it may take a few years for it to happen. kubeadm users tend to watch for warnings in the logs, so hopefully most users have adapted to paths with URL prefix by now.

Reply:
cc @SataQiu what is your take on this one?

Reply:
Just attend Jordon's talk in the contributor summit. We should avoid `action-required` if we can.

/close
if you guys think this may hurt users, that is what we don't want.

Reply:
<img width="977" alt="企业微信截图_d206add4-b220-4ea6-a38f-f8d5e396da8c" src="https://user-images.githubusercontent.com/2010320/233154455-08bc72d4-c9b9-4251-9343-b67cbff66b18.png">

Reply:
if / when the kubelet does the change we can make the change in kubeadm as well so that an action-required is for both components in the same release.
if the kubelet never does that we can just keep the kubeadm warning.